src/ui/common/src/components/workflows/ReactFlowCanvas.tsx

```diff
@@ -44,11 +52,148 @@ const ReactFlowCanvas: React.FC<ReactFlowCanvasProps> = ({
     setTimeout(fitView, 200);
   }, [openSideSheetState]);
 
+  const checkOpNodes = [];
+  const boolArtifactNodes = [];
+
+  const metricOpNodes = [];
+  const metricArtifactNodes = [];
+
+  // first find all check operators.
+  if (dagPositionState.result) {
+    const nodes = dagPositionState.result.nodes;
```

Comment:
let's put this into a helper function called something like 'collapse nodes'?

Reply:
good catch, will add this :)